### PR TITLE
(#3200) - Use a request polyfill for xhr

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

**Comment by daleharvey** [comment]
Actually fairly surprised here, this passes all but 3 tests in firefox at least

**Comment by daleharvey** [comment]
All but 2, those 2 are intermittent, really hard to get reproducing in a sane way, may be related to the deletion race

**Comment by daleharvey** [comment]
Ok so this should be passing tests, not implementing the cache busting made it break in curious and fun ways, one issue is that it adds 16K to the minified build size despite no extra modules and almost the same code

**Comment by daleharvey** [comment]
Ok found it, there was a `new Buffer` in there

So this doesnt technically change much, request-browser is pretty much what used to be in `ajax-browser.js`, it just simplifies the dependency tree so the xhr logic is behind request, if there is a better request browser polyfill that is compatible where we need it and a smaller build size then it can be replaced much easier than before

Will likely want to publish request-browser at some point but I would like to get android testing then hopefully remove createBlob before worrying about that

**Comment by nolanlawson** [comment]
Android 4.0-4.3 is really crucial to manually test before merging any XHR changes. I remember there being a bizarre http bug in that version, and sadly [4.0-4.3 is still a big chunk of Android users](https://developer.android.com/about/dashboards/index.html).

**Comment by daleharvey** [comment]
ok think I have found the last of the issues for this, all worked fine except attatchments broke in various ways in various browsers

**Comment by daleharvey** [comment]
Ok this is passing in every local browser I can test, should be good to go now

**Comment by nolanlawson** [comment]
I'll manually test in android 4.3

**Comment by daleharvey** [comment]
I have been between trying to manually test and get https://github.com/pouchdb/pouchdb/pull/3425 up and running

File: tests/integration/test.attachments.js [hunk 1]
```diff
@@ -1749,7 +1749,7 @@ adapters.forEach(function (adapter) {
                 cache: true,
                 binary: true
               }, function (err, res) {
-                if (err && err.status === 405) {
+                if (err && err.status === 500) {
```
Comment (nolanlawson) [inline tests/integration/test.attachments.js]: ^ what's the deal here? I know firefox had fixed the 405 bug in nightlies, but for older versions of firefox, won't this cause us to use base64 when actually blobs are okay?

**Comment by nolanlawson** [comment]
Nice, min+gz size only increased from 46035 to 46132.

File: tests/integration/test.attachments.js (hunk 1 above)
Comment (daleharvey) [inline tests/integration/test.attachments.js]: As per the comment in the test, this check only to skip the tests in firefox, we previously had done https://github.com/pouchdb/pouchdb/blob/master/lib/deps/ajax-browser.js#L80 to always force a 405 which was causing various ways xhr could fail so misreport as cors problems, afaik it should affect blob detection

We are testing in fx 31 in travis which is older than we support anyway

Comment (nolanlawson) [inline tests/integration/test.attachments.js]: okay, I remember now. lgtm

**Comment by nolanlawson** [comment]
100% in android 4.2.2. ship it :+1:

File: tests/integration/test.attachments.js (hunk 1 above)
Comment (nolanlawson) [inline tests/integration/test.attachments.js]: For the record, it seems Firefox 35+ has the blob fix, so it doesn't report any kind of error.

As for previous versions of Firefox, I'm glad we're testing it, but to be fair, our tests won't capture if we get a false negative and revert to base64 when actually blob would be appropriate. But given that base64 works universally, and that Firefox is so good about auto-updating, I'm not really concerned.

**Comment by nolanlawson** [comment]
a82c273d5e96146b366ce55780337d629eb169ba

**Pull request closed without merging**